src/System/Components/RouterLink.tsx

```diff
@@ -52,7 +52,11 @@ export const RouterLink: React.FC<React.PropsWithChildren<React.PropsWithChildre
     const isPrefetchOnEnterEnabled =
       isPrefetchOnEnterEnabledLoggedIn || isPrefetchOnEnterEnabledLoggedOut
 
-    const { prefetch } = usePrefetchRoute(to as string)
+    // When a prefetch is completed, update the route to include a `is_prefetched=true` query param.
+    const [isPrefetched, setIsPrefetched] = React.useState(false)
+    const { prefetch } = usePrefetchRoute(to as string, () =>
+      setIsPrefetched(true)
+    )
```

Comment:
we should update this API a bit now that we'r accepting additional props and follow object / relay / etc forms:

```tsx
const { prefetch } = usePrefetchRoute({
  to,
  onComplete: () => {
    setIsPrefetched(true)
  }
)
```